### PR TITLE
Fix `@wordpress/comment-case` ESlint errors but without adding the disable-rule pragma

### PR DESCRIPTION
**Pull request by opr**
## Description

This PR is a cleanup of code comments that don't conform to the new `@wordpress/comment-case` rule.

It is Related to #36866 but the comments  such as `eslint-disable-next-line` etc. are removed.

## How has this been tested?

It is based off #36866 and I manually inspected every file there, this PR just removes the comments that disable the ESLint rule, I didn't test all of the other files.

## Types of changes

This PR only changes comments, no functionality would be affected.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by gziolo** [comment]
There is one failing JS unit test:

Screenshot 2021-11-30 at 16 30 01

I bet it's caused by the changes in the source file used in the test fixture. It's probably enough to regenerate the test snapshot.

File: packages/autop/src/test/index.test.js
Comment (cbravobernal) [inline packages/autop/src/test/index.test.js:137]: Do we have to add a dot at the end of this line?

File: packages/block-editor/src/components/alignment-control/test/index.js
Comment (cbravobernal) [inline packages/block-editor/src/components/alignment-control/test/index.js:78]: Seems that is not adding a dot after symbols > )

File: packages/block-editor/src/components/inserter/menu.native.js
Comment (cbravobernal) [inline packages/block-editor/src/components/inserter/menu.native.js]: // If should replace but didn't insert any block

File: packages/components/src/color-picker/test/index.js
Comment (cbravobernal) [inline packages/components/src/color-picker/test/index.js]: Sometimes seems that did not got the uppercase rule 🤔

File: packages/dependency-extraction-webpack-plugin/test/__snapshots__/build.js.snap
Comment (cbravobernal) [inline packages/dependency-extraction-webpack-plugin/test/__snapshots__/build.js.snap]: No sure what will happen here, it's a different version.

**Review by cbravobernal — APPROVED** [review]
It seems that sometimes is missing adding any extra space, or not doing the uppercase, but overall is a great work.

We could have an issue with this snapshot, is expecting an output with different version.

packages/dependency-extraction-webpack-plugin/test/__snapshots__/build.js.snap

**Comment by cbravobernal** [comment]
> There is one failing JS unit test:
> 
> Screenshot 2021-11-30 at 16 30 01
> 
> I bet it's caused by the changes in the source file used in the test fixture. It's probably enough to regenerate the test snapshot.

https://github.com/WordPress/gutenberg/pull/37006#discussion_r759462636

File: packages/block-editor/src/components/inserter/menu.native.js
Comment (opr) [reply packages/block-editor/src/components/inserter/menu.native.js]: Fixed in https://github.com/WordPress/gutenberg/pull/34964/commits/59666727f6ec81241d4eb6977cc63d2feb3dddb2 - turns out I was checking for common words incorrectly

File: packages/autop/src/test/index.test.js
Comment (opr) [reply packages/autop/src/test/index.test.js:137]: I'm not sure, I purposely stopped it from considering lines that ended in non-word characters, do you think I should remove this exception and make periods mandatory even after non-word chars?

File: packages/components/src/color-picker/test/index.js
Comment (opr) [reply packages/components/src/color-picker/test/index.js]: Ahhh this is because `a` is an identifier on the next line, and the rule will not enforce it if the comment starts with any identifiers on the next or previous lines. I'll make it check for an identifier and a space, that will solve this.

File: packages/block-editor/src/components/alignment-control/test/index.js
Comment (opr) [reply packages/block-editor/src/components/alignment-control/test/index.js:78]: See  https://github.com/WordPress/gutenberg/pull/37006#discussion_r759472277

File: packages/components/src/color-picker/test/index.js
Comment (opr) [reply packages/components/src/color-picker/test/index.js]: Fixed in 0b0145f28a545c65c28c9a74bde5155de8c8090a

**Comment by opr** [comment]
> It seems that sometimes is missing adding any extra space, or not doing the uppercase, but overall is a great work.
> 
> We could have an issue with this snapshot, is expecting an output with different version.
> 
> packages/dependency-extraction-webpack-plugin/test/**snapshots**/build.js.snap

Why would the version have changed? Surely my comment changes wouldn't have caused this, and it seems only the version has updated. I would be keen to have someone who knows more about this particular area of code to take a look!

**Comment by opr** [comment]
> It seems that sometimes is missing adding any extra space, or not doing the uppercase, but overall is a great work.

Can you point me to a file where this has happened please? I'll take a look!

**Comment by cbravobernal** [comment]
> > It seems that sometimes is missing adding any extra space, or not doing the uppercase, but overall is a great work.
> 
> Can you point me to a file where this has happened please? I'll take a look!

I will try to 😆 . It will be hard to remember after 500 files!

**Comment by cbravobernal** [comment]
>Can you point me to a file where this has happened please? I'll take a look!

https://github.com/opr/gutenberg/blob/21da99853d2e1ae451ae0efb9c600abd89cf418d/packages/components/src/higher-order/with-focus-outside/index.js#L1

https://github.com/opr/gutenberg/blob/21da99853d2e1ae451ae0efb9c600abd89cf418d/packages/components/src/confirm-dialog/test/index.js#L114

https://github.com/opr/gutenberg/blob/21da99853d2e1ae451ae0efb9c600abd89cf418d/packages/components/src/confirm-dialog/test/index.js#L249

File: packages/autop/src/test/index.test.js
Comment (gziolo) [reply packages/autop/src/test/index.test.js:137]: I think it's fine to leave it as is. The simple solution would bo to end a sentence with a word 😄

Comment (gziolo) [reply packages/autop/src/test/index.test.js:137]: Well, can we still fix it manually without conflicting with the rule fixer?

File: packages/block-editor/src/components/block-list/block-selection-button.native.js
Comment (gziolo) [inline packages/block-editor/src/components/block-list/block-selection-button.native.js:41]: I think we have a different issue here. This is a single line comment so it should start with `//`.

File: packages/block-editor/src/components/block-switcher/use-transformed-patterns.js
Comment (gziolo) [inline packages/block-editor/src/components/block-switcher/use-transformed-patterns.js]: This comment is useless:
```suggestion
```

File: packages/block-editor/src/components/list-view/block.js
Comment (gziolo) [inline packages/block-editor/src/components/list-view/block.js:136]: Any ideas why this line isn't capitalized, nor it ends with a dot?

File: packages/block-editor/src/utils/transform-styles/ast/parse.js
Comment (gziolo) [inline packages/block-editor/src/utils/transform-styles/ast/parse.js:243]: It doesn't start with an upper case. Is it because it's also a variable in the scope? Anyway, let's be honest that this comment is redundant similar to other single words in the same file.

File: packages/components/src/input-control/input-base.tsx
Comment (gziolo) [inline packages/components/src/input-control/input-base.tsx:75]: This raises a flag, but I guess it's only an explanation for the expected TypeScript error.

File: packages/dependency-extraction-webpack-plugin/test/__snapshots__/build.js.snap
Comment (gziolo) [reply packages/dependency-extraction-webpack-plugin/test/__snapshots__/build.js.snap]: It's all good because the content of the source file has changed:

```diff
- /* Silence is golden */
+ /* Silence is golden. */
```

I assume we don't use the minification for the generated test output so the calculated has has changed.

**Review by gziolo — APPROVED** [review]
This is great. I scanned all files. It took me a lot of time. I didn't find anything concerning that could block this PR 🎉 

Unfortunately, as noted by @noisysocks in https://github.com/WordPress/gutenberg/pull/36866#issuecomment-983136127 we will have until January to land all those changes so they don't impact negatively the release process of the next major WordPress version. @opr, I hope you don't mind keeping this PR up to date with the latest changes in `trunk` for so long. I bet it might become a bit painful as time progresses.

**Comment by gziolo** [comment]
@op, we would love to land it. Do you need any help with bringing it up to date?

**Comment by opr** [comment]
> @opr, we would love to land it. Do you need any help with bringing it up to date?

Hey I'll take a look into getting this up to date next week (w/c 14th February)! 🕺🏼

File: .eslintrc.js
Comment (gziolo) [inline .eslintrc.js]: It looks like a merge issue.
```suggestion
```

**Review by gziolo — APPROVED** [review]
This looks great. I'm excited that we can finally land this PR. Great work @opr! 👏🏻

**Pull request merged**
